### PR TITLE
Implement AdventureSettings & PlayerSkin packet & Sandstone

### PR DESCRIPTION
**Pull request by Suremeo**
Changes:

- Handle the AdventureSettings packet
- Flying related functions
- Added player scaling
- Added updating player skin
- Handle the PlayerSkin packet (So other players see when you change your skin ingame)
- Implemented Sandstone

### CODE REVIEW

**Comment by Eren5960** [comment]
player.go#L359 must be start with SetScale ...

**Comment by Eren5960** [comment]
no problem, run `go fmt ./...` in base dragonfly path and fix formatting

**Comment by Suremeo** [comment]
> no problem, run `go fmt ./...` in base dragonfly path and fix the formatting errors

Done

File: dragonfly/entity/state/state.go [hunk 1]
```diff
@@ -50,6 +50,11 @@ type UsingItem struct{}
 // OnFire makes an entity show itself as on fire.
 type OnFire struct{}
 
+// Scale sets the scale of the players geometry
```
Comment (DaPigGuy) [inline dragonfly/entity/state/state.go:53]: Missing punctuation

File: dragonfly/player/player.go [hunk 2]
```diff
@@ -354,6 +356,17 @@ func (p *Player) Speed() float64 {
 	return p.speed.Load()
 }
 
+// SetScale sets the scale of the players geometry
```
Comment (DaPigGuy) [inline dragonfly/player/player.go:359]: Missing punctuation

File: dragonfly/player/player.go [hunk 3]
```diff
@@ -354,6 +356,17 @@ func (p *Player) Speed() float64 {
...
+func (p *Player) SetScale(scale float64) {
+	p.scale.Store(scale)
+	p.updateState()
+}
+
+// Scale returns the current scale of the player
```
Comment (DaPigGuy) [inline dragonfly/player/player.go:365]: Missing punctuation.

File: dragonfly/player/player.go [hunk 4]
```diff
@@ -1718,6 +1731,9 @@ func (p *Player) State() (s []state.State) {
 	if p.OnFireDuration() > 0 {
 		s = append(s, state.OnFire{})
 	}
+	if p.scale.Load() >= 1 {
```
Comment (DaPigGuy) [inline dragonfly/player/player.go:1734]: Scale less than 1 won't be sent

**Review by DaPigGuy — CHANGES_REQUESTED** [review]

File: dragonfly/entity/state/state.go (hunk 1 above)
Comment (Suremeo) [reply dragonfly/entity/state/state.go]: I don't see any issue with the punctuation on this one

Comment (T14Raptor) [reply dragonfly/entity/state/state.go]: Forgot a period

Comment (Suremeo) [reply dragonfly/entity/state/state.go]: Ah, frogot documentation is supposed to have them, fixed on all files now.

**Comment by Suremeo** [comment]
Added lots of other stuff to this pull request

Changes:

- Handle the AdventureSettings packet
- Flying related functions
- Added player scaling
- Added updating player skin
- Handle the PlayerSkin packet (So other players see when you change your skin ingame)

**Comment by Sandertv** [comment]
Please make sure to keep separate changes in a different PR the next time. This makes it very difficult to review and get merged.

**Comment by Suremeo** [comment]
> Please make sure to keep separate changes in a different PR the next time. This makes it very difficult to review and get merged.

Alright, will do

**Comment by T14Raptor** [comment]
Closing in favor of d3a8a5610c33055a9efa5b9293d0ef840e2ec5af, f255865bf8b9515a722e17653005f4cedc9ffcef, 4e0e67afd2aa2b79b3a3c20db3f37856f165286d, etc.

**Pull request closed without merging**